stored_requests/backends/db_fetcher/postgres.go

```diff
@@ -18,6 +18,15 @@ func NewPostgres(cfg *config.PostgresConfig) (stored_requests.Fetcher, error) {
 		return nil, err
 	}
 
+	switch qtype {
```

Comment:
A few issues with this...

1. As writte, this will open up two SQL connections.
2. Magic strings are error-prone.

There are a couple ways you could make it more robust... but I see a few options:

- Create the SQL connection in `pbs_light.go`, and make this method accept `(sql, queryMaker)` as its two args.
- Make this `NewPostgres(cfg) (openrtbFetcher Fetcher, ampFetcher Fetcher, error)`. `NewFetcher` in `pbs_light.go` would have to return both at once as well.

Reply:
I liked the case as it is easy to expand in case of future different types of ID queries. But the magic strings is a valid point.

I am not so sure that 2 SQL connections is a real issue. The postgresql library could be implemented in one of two ways:
 1. One DB connection means one DB connection. That means an ID fetcher request will block all other from other go routines.
2. The postgresql library maintains a connection pool, and expands/contracts it as necessary. This means we may have two pools instead of one (depends on if the library is smart enough to merge similar pools)

In the case of 1, we will only be able to scale to the size of the ID cache we have in front of the database, beyond that performance will tank. In the case of 2, once we pass the cache size, performance will drop to no cache levels until we overrun the capacity of the database. This is a much more graceful degradation.

Reply:
If we move to one of your suggestions, I am leaning towards the pbs_light.go solution, it is seems cleaner to expand. Or pushing the query type into an argument for FetchRequests, and having it pick the proper query to use.